django_keycloak/middleware.py

```diff
@@ -42,24 +79,28 @@ def append_user_info_to_request(self, request, token):
                 user.email = user_info.get("email")
                 user.save()
 
-        except get_user_model().DoesNotExist:
-            user = get_user_model().objects.create_from_token(token)
+        except User.DoesNotExist:
+            user = User.objects.create_from_token(token)
+
+        # Add the local user to request
         request.user = user
 
         return request
 
     @staticmethod
-    def is_auth_header_missing(request):
+    def has_auth_header(request) -> bool:
         """Check if exists an authentication header in the HTTP request"""
-        return "HTTP_AUTHORIZATION" not in request.META
+        return "HTTP_AUTHORIZATION" in request.META
 
     @staticmethod
     def get_token(request):
-        """Get the token from the HTTP request"""
+        """
+        Get the token from the HTTP request
+        """
         auth_header = request.META.get("HTTP_AUTHORIZATION").split()
-        if len(auth_header) == 2:
+        if len(auth_header) > 1:
```

Comment:
Is there a case for a valid HTTP header where it is not 2?

Reply:
For a strictly valid no, I guess it is redundant.

Reply:
You pass `Bearer mytoken x`. It will read it as `mytoken`, the old code would read it as `Bearer`. Don't know if this has any deeper consequences. What do you think?

Reply:
Cool, then we can remove `get_token()`

Reply:
Maybe we can add a condition to check if user correctly formatted the header?
Along the lines of: 
```python
if not (auth_type and value) or auth_type[0].lower() not in ["basic", "bearer"]:
            raise SomeError
```

Reply:
I'm for being strict with the user data in the cases that we handle, but should also not block the usage of other auth middleware (digest, hawk, aws) that also uses the auth header:

https://www.loginradius.com/blog/engineering/everything-you-want-to-know-about-authorization-headers/

Reply:
You are right, forgot about other middlewares 👍

Reply:
Also, I have removed graphene middleware. Our middleware is now a single class.